test/CodeGen/AVR/sign-extension.ll

```diff
@@ -0,0 +1,11 @@
+; RUN: llc -march=avr < %s | FileCheck %s
+
+define i64 @sign_extended(i1) {
+; CHECK-LABEL: entry-block
+entry-block:
+  %1 = sext i1 %0 to i8
+  %2 = sext i8 %1 to i16
```

Comment:
I wanted to give a bit more thought to see if I could come up with a nicer solution than the mess of assembly, but if I fail at that, I'll merge :-)